tools/rostopic/src/rostopic/__init__.py

```diff
@@ -1202,14 +1227,7 @@ def _rostopic_list(topic, verbose=False,
     
     master = rosgraph.Master('/rostopic')
     try:
-        state = master.getSystemState()
-
-        pubs, subs, _ = state
-        if topic:
-            # filter based on topic
-            topic_ns = rosgraph.names.make_global_ns(topic)        
-            subs = (x for x in subs if x[0] == topic or x[0].startswith(topic_ns))
-            pubs = (x for x in pubs if x[0] == topic or x[0].startswith(topic_ns))
+        pubs, subs = get_topic_list()
```

Comment:
The new function returns different information than `getSystemState` before. I doubt that all following code continues to work as is.